src/domain/safe-apps/entities/schemas/__tests__/safe-app.schema.spec.ts

```diff
@@ -64,39 +68,59 @@ describe('SafeAppSchema', () => {
     );
   });
 
-  it('should validate nested socialProfile and provider urls', () => {
+  it.each(Object.values(SafeAppSocialProfilePlatforms))(
+    'should validate %s nested socialProfile and provider urls',
+    (platform) => {
+      const safeApp = safeAppBuilder()
+        .with('socialProfiles', [
+          safeAppSocialProfileBuilder().with('platform', platform).build(),
+          safeAppSocialProfileBuilder()
+            .with('platform', platform)
+            .with('url', faker.string.alphanumeric())
+            .build(),
+        ])
+        .with(
+          'provider',
+          safeAppProviderBuilder()
+            .with('url', faker.string.alphanumeric())
+            .build(),
+        )
+        .build();
+
+      const result = SafeAppSchema.safeParse(safeApp);
+
+      expect(!result.success && result.error).toStrictEqual(
+        new ZodError([
+          {
+            validation: 'url',
+            code: 'invalid_string',
+            message: 'Invalid url',
+            path: ['socialProfiles', 1, 'url'],
+          },
+          {
+            validation: 'url',
+            code: 'invalid_string',
+            message: 'Invalid url',
+            path: ['provider', 'url'],
+          },
+        ]),
+      );
+    },
+  );
+
+  it('should fallback to UNKNOWN nested socialProfile', () => {
     const safeApp = safeAppBuilder()
       .with('socialProfiles', [
-        safeAppSocialProfileBuilder().build(),
         safeAppSocialProfileBuilder()
-          .with('url', faker.string.alphanumeric())
+          .with('platform', 'invalid' as SafeAppSocialProfile['platform'])
           .build(),
       ])
-      .with(
-        'provider',
-        safeAppProviderBuilder()
-          .with('url', faker.string.alphanumeric())
-          .build(),
-      )
       .build();
 
     const result = SafeAppSchema.safeParse(safeApp);
```

Comment:
We are testing that our validation doesn't throw if a value other than `SafeAppSocialProfilePlatforms` is received. [It should assign `UNKNOWN`](https://github.com/safe-global/safe-client-gateway/blob/16eb3b7166872600260f835e4e9ba0306b7e7451/src/domain/safe-apps/entities/schemas/safe-app.schema.ts#L27). This matches other areas in our codebase, e.g. [for `Order`s](https://github.com/safe-global/safe-client-gateway/blob/1a84d6b29411b6d56ac3235aebeebeeb50be44e4/src/domain/swaps/entities/order.entity.ts#L62).

Reply:
As discussed, Since the value is hard-coded and not dynamically assigned we could trust that if falls back to `UNKNOWN` but, Here the focus of the test is to ensure that we have the fallback handler in place.